python-checks/src/main/java/org/sonar/python/checks/hotspots/LoggersConfigurationCheck.java

```diff
@@ -0,0 +1,93 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.hotspots;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.python.PythonSubscriptionCheck;
+import org.sonar.python.SubscriptionContext;
+import org.sonar.python.api.tree.HasSymbol;
+import org.sonar.python.api.tree.PyArgListTree;
+import org.sonar.python.api.tree.PyArgumentTree;
+import org.sonar.python.api.tree.PyAssignmentStatementTree;
+import org.sonar.python.api.tree.PyCallExpressionTree;
+import org.sonar.python.api.tree.PyClassDefTree;
+import org.sonar.python.api.tree.Tree;
+import org.sonar.python.semantic.TreeSymbol;
+
+@Rule(key = "S4792")
+public class LoggersConfigurationCheck extends PythonSubscriptionCheck {
+
+  private static final Set<String> FUNCTIONS_TO_CHECK = new HashSet<>(Arrays.asList(
+    "logging.basicConfig",
+    "logging.disable",
+    "logging.setLoggerClass",
+    "logging.config.fileConfig",
+    "logging.config.dictConfig"));
+
+  private static final Set<String> LOGGERS_CLASSES = new HashSet<>(Arrays.asList(
+    "logging.Logger",
+    "logging.Handler",
+    "logging.Filter"));
+
+  private static final String MESSAGE = "Make sure that this logger's configuration is safe.";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, ctx -> {
+      PyCallExpressionTree callExpressionTree = (PyCallExpressionTree) ctx.syntaxNode();
+      TreeSymbol symbol = callExpressionTree.calleeSymbol();
+      if (symbol != null && FUNCTIONS_TO_CHECK.contains(symbol.fullyQualifiedName())) {
+        ctx.addIssue(callExpressionTree, MESSAGE);
+      }
+    });
+
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_STMT, ctx -> isSettingLastResort(ctx, (PyAssignmentStatementTree) ctx.syntaxNode()));
+
+    context.registerSyntaxNodeConsumer(Tree.Kind.CLASSDEF, ctx -> isClassExtendingLogger(ctx, (PyClassDefTree) ctx.syntaxNode()));
+  }
+
+  private static void isClassExtendingLogger(SubscriptionContext ctx, PyClassDefTree classDef) {
+    PyArgListTree argList = classDef.args();
+    if (argList != null) {
+      argList.arguments().stream()
+        .map(PyArgumentTree::expression)
+        .filter(expr -> expr instanceof HasSymbol && ((HasSymbol) expr).symbol() != null)
+        .filter(expr -> LOGGERS_CLASSES.contains(((HasSymbol) expr).symbol().fullyQualifiedName()))
+        .forEach(expr -> ctx.addIssue(expr, MESSAGE));
+    }
+  }
+
+  // check if logging.lastResort is being set
+  private static void isSettingLastResort(SubscriptionContext ctx, PyAssignmentStatementTree assignment) {
+    assignment.lhsExpressions().stream()
+      .flatMap(exprList -> exprList.expressions().stream())
+      .forEach(expr -> {
+        if (expr instanceof HasSymbol) {
```

Comment:
This is past the scope of this PR but there is some design flaw in terms of OOP in the fact that we have to check if an expr is a "hasSymbol" instance via an instance of AND then check if symbol is null. 
Why not having a symbol method on every expression for instance and returning a "NASymbol" or a "unknown" in the different cases.

Reply:
I agree this is a bit cumbersome, I will take this point in a separate PR